src/api/createDeviceMessageHandlingSystem/src/v1/steps.py

```diff
@@ -0,0 +1,125 @@
+from http import HTTPStatus
+
+from domain.api.common_steps.general import parse_event_body
+from domain.api.common_steps.read_product import (
+    parse_path_params,
+    read_product,
+    read_product_team,
+)
+from domain.core.cpm_product.v1 import CpmProduct
+from domain.core.device.v3 import Device
+from domain.core.error import InvalidSpineMhsResponse, NotEprProductError
+from domain.core.product_key.v1 import ProductKeyType
+from domain.core.questionnaire.v3 import Questionnaire, QuestionnaireResponse
+from domain.repository.device_repository.v3 import DeviceRepository
+from domain.repository.questionnaire_repository.v2 import QuestionnaireRepository
+from domain.repository.questionnaire_repository.v2.questionnaires import (
+    QuestionnaireInstance,
+)
+from domain.request_models.v1 import CreateMhsDeviceIncomingParams
+from domain.response.validation_errors import mark_validation_errors_as_inbound
+
+
+@mark_validation_errors_as_inbound
+def parse_mhs_device_payload(data, cache) -> Device:
+    payload: dict = data[parse_event_body]
+    return CreateMhsDeviceIncomingParams(**payload)
+
+
+def get_party_key(data, cache) -> str:
+    product: CpmProduct = data[read_product]
+    party_keys = [
+        key.key_value
+        for key in product.keys
+        if key.key_type is ProductKeyType.PARTY_KEY
+    ]
```

Comment:
Bit of a subtlety with zero impact here, but switching from `[]` to `()` here is (generally) more efficient because line 37 will fail faster

```suggestion
    party_keys = (
        key.key_value
        for key in product.keys
        if key.key_type is ProductKeyType.PARTY_KEY
    )
```

feel free to ignore...

Reply:
Ah yeah, I've changed it in createDeviceReferenceDataMessageSet too - although thinking about it now, that step can probably be another common product step